btree/btree.c

```diff
@@ -4509,10 +4514,30 @@ void m0_btree_put_credit(const struct m0_btree  *tree,
 	m0_be_tx_credit_mac(accum, &cred, nr);
 }
 
-/**
- * This function will calculate credits required to perform @nr delete KV
- * operations and it will add those credits to @accum.
- */
+void m0_btree_put_credit2(const struct m0_btree_type *type,
```

Comment:
In current credit calculation, we are using MAX_HEIGHT, but btree user may provide us exact height so we can optimize credit calculations.   In old btree,  there are functions m0_be_btree_insert_credit2, m0_be_btree_delete_credit2, etc. which
used btree height provided by user, We can do the similar thing, as if you are creating dummy tree that means height is not known so we can set height to MAX_height, and call this function if btree pointer is not provided to us.

Reply:
@RadhaGulhane13  - Our current btree credit calculation functions do not take the parameter to use either MAX_HEIGHT or user provided height. If we ever start to support this then at that time we should change the _credit2() functions - in other words that will be a different task.

Reply:
okay.